backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/JWTgenerator.java

```diff
@@ -0,0 +1,74 @@
+package org.molgenis.emx2;
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.MACVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import java.security.SecureRandom;
+import java.util.Date;
+
+public class JWTgenerator {
+  private static byte[] sharedSecret;
+  private static MACSigner signer;
+  private static JWSVerifier verifier;
+
+  private static void init() {
+    try {
+      // Generate random 256-bit (32-byte) shared secret
+      sharedSecret = new byte[32];
+      new SecureRandom().nextBytes(sharedSecret);
+
+      // Create HMAC signer
+      signer = new MACSigner(sharedSecret);
+      verifier = new MACVerifier(sharedSecret);
+    } catch (Exception kle) {
+      throw new MolgenisException(kle.getMessage());
+    }
+  }
+
+  // create token
+  public static String createTokenForUser(String user, int minutes) {
+    try {
+      if (signer == null) {
+        init();
+      }
+      // Prepare JWT with claims set
+      JWTClaimsSet claimsSet =
+          new JWTClaimsSet.Builder()
+              .subject(user)
+              .expirationTime(new Date(new Date().getTime() + minutes * 1000))
```

Comment:
Indeed I want to detach it. So in case of openid connect is used to sign in then we probably need another endpoint to retrieve the token (i.e. not via the sign in event).]

I indeed want to use this only for short lived tokens.  For long running tokens we might want to use a mechanism that either uses token from the openid connect OR that we create tokens via admin, seperately from openid connect. I would like to discuss this in a later pull request :-)